### ISSUE TITLE
Duplicate URLs for articles

### ISSUE THREAD

**Issue by patbl** [tog22/eaforum#54]
Sometimes the same article appears twice in the RSS feed. It looks as though it has something to do with people revising articles. But I've never seen the same article more than twice, but I imagine that people have revised articles more than twice.

Here are a couple of examples:

<img width="1128" alt="screenshot 2016-02-24 18 13 12" src="https://cloud.githubusercontent.com/assets/1735266/13304557/51d623c4-db23-11e5-860e-5a1914571364.png">

http://effective-altruism.com/ea/tj/allkind/
http://effective-altruism.com/ea/tg/allkind/

<img width="1141" alt="screenshot 2016-02-24 18 13 06" src="https://cloud.githubusercontent.com/assets/1735266/13304579/7d4269f0-db23-11e5-900d-a3acd4a91154.png">

http://effective-altruism.com/ea/tn/should_we_work_on_taxation_of_the_very_rich/
http://effective-altruism.com/ea/tn/should_effective_altruists_work_on_taxation_of/

It looks as though the "Allkind" article may have a separate issue from that of the taxation article. Both taxation URLs seem to point to the same article (the "/ea/tn/" bits of the URLs are identical). But the "Allkind" URLs have different bits ("tj" vs. "tg"), and the number of upvotes is different.

Most users probably don't notice or aren't bothered by this problem. Having duplicate URLs would hurt the search-engine rankings of the duplicate articles (and possibly the ranking of the site as a whole). I don't know much about SEO, but my guess is that the effect is small.

The lack of a canonical URL for an article might be indicative of other problems, so I think this is worth investigating.

**Comment by tog22** [comment]
Might that be this issue? https://dotimpact.slack.com/archives/p-forum/p1455547310000002

**Comment by peterhurford** [comment]
@patbl @tog22 Did this ever get fixed?

**Comment by patbl** [comment]
No.

**Comment by tog22** [comment]
@patbl based on what you've seen does it appear that it might be the issue I linked to? Pasting this here for when the Slack archive expires:

Alasdair Pearce [6:41 AM] 
Bug/question - I keep getting posts in my rss feed that when I go to the main site are not there / not there yet. Like http://effective-altruism.com/ea/tm/saving_the_human_rights_act_in_the_uk/ and a few days ago what I think was a draft of tyler's Allkind essay. Is there some sort of bug with how the rss feed for the forum works?
TL;DR  Could the pooled expertise of the EA community overcome the weakness of evidence on both sides of an upcoming British policy reform concerning human r

Tom Ash [8:57 AM] 
I suspect that was someone publishing and then unpublished (ie. switching back to a draft). Could you try doing that with a post and see if the prob arises?

**Comment by patbl** [comment]
A little bit off topic, but is it the case that links to Slack conversations stop working? They say that on the free plan you can "Search and browse 10k most recent messages." They don't say whether you can link to older messages.

**Comment by tog22** [comment]
That's a very good question - if you checked when you had the time and inclination that'd be handy 😍

**Comment by patbl** [comment]
There's definitely a bug here, but there's also a very confusing UI that causes people to accidentally post drafts.

There are two buttons, "Submit" and "Save and continue," that as far as I can tell do almost the same thing. The only difference is that "Submit" redirects you to the article URL, while "Save and continue" keeps you on the "Submit article" page.

![confusing-ui](https://cloud.githubusercontent.com/assets/1735266/15101081/5f635276-153b-11e6-81d0-13bbfea3a2bc.png)

I would expect "Save and continue" to simply save your article, not make it public. But to control that behavior, you have to use a drop-down menu that typically has only two options: "Drafts for so-and-so" and "EA Forum." And it seems that the drop-down menu defaults to "EA Forum." So if you don't notice that drop-down menu, or you don't understand what it does, you're likely to publish an article before it's ready.

Here are some ideas for a better UI:
- Get rid of the drop-down menu.
- Have two buttons, "Save" and "Publish." It might be good to display a confirmation dialog when someone clicks "Publish" in case they fat-finger it.
- After the article is published, there should be two buttons, "Update" and "Unpublish."

There should probably be a "Delete" button on the edit page as well. It's weird that you can only delete it by going to the public URL.

It would be a bit of work to make these changes, and maybe they're not worth it. But I thought I'd at least record my thoughts here.

**Comment by patbl** [comment]
For articles that have the same identifier (`tw` below) but different slugs, you can reproduce the bug by changing the title of the article after it's published.

http://effective-altruism.com/ea/tw/the_most_persuasive_writing_neutrally_surveys/
http://effective-altruism.com/ea/tw/dont_write_the_case_for_one_side_survey_both/

The solution is probably to always use the original slug even if the title changes.

I'm not sure what causes the other issue, when there are multiple versions of an article that have different identifiers. But this seems significantly less common (although more serious).

**Comment by tog22** [comment]
Agree that this is bad UX. Thanks for bringing it up and suggesting
changes! I think they're good, so please create a github issue for them,
and just unassign yourself from it if you can later tell that you're not
going to work on it. If you do, tag Alex and ask if he's up for doing so.
:)

**Comment by arichard4** [comment]
Regular reddit appears to have solved the title issue by making them non-modifiable: "Titles, however, are not modifiable once submitted, get those right. If you need to change a title, delete your post and re-submit it."

https://www.reddit.com/wiki/submitting

**Comment by patbl** [comment]
The change I made should prevent a new URL from being created when an article title is updated. But the change doesn't affect the other bug (supposing it is a bug, and not just strange behavior by users) wherein the same article has more than one ID.

**Comment by peterhurford** [comment]
@patbl Thanks for the PR. Should we close this issue now?

**Comment by patbl** [comment]
Yes, it looks as though the other issue could well have been caused by people deleting their accounts and re-posting the same article under another account.

http://effective-altruism.com/ea/tg/allkind/
http://effective-altruism.com/ea/tj/allkind/
